diff --git a/node_modules/c2pa/dist/c2pa.esm.js b/node_modules/c2pa/dist/c2pa.esm.js
index 6221bf9..a1e3525 100644
--- a/node_modules/c2pa/dist/c2pa.esm.js
+++ b/node_modules/c2pa/dist/c2pa.esm.js
@@ -21293,17 +21293,24 @@ function createWorkerPool(config) {
  */
 const dbg$4 = debug('c2pa:workers');
 async function createPoolWrapper(config) {
-    const res = await fetch(config.scriptSrc);
-    if (!res.ok)
-        throw new InvalidWorkerSourceError(config.scriptSrc, res);
-    const src = await res.text();
-    // @TODO: check subresource integrity
-    dbg$4('Fetched worker from %s (%d bytes)', config.scriptSrc, src.length);
-    const workerBlob = new Blob([src], { type: 'application/javascript' });
-    const workerUrl = URL.createObjectURL(workerBlob);
+	/*
+
+		Firefox v3 extension CSP restrictions will not allow a worker to be instantiated from a blob source.
+		Since we are including this lib as a static file, the transformation from URL to blob is unnecessary.
+		We instantiate the worker the extension supplied URL directly.
+
+	*/
+    // const res = await fetch(config.scriptSrc);
+    // if (!res.ok)
+    //     throw new InvalidWorkerSourceError(config.scriptSrc, res);
+    // const src = await res.text();
+    // // @TODO: check subresource integrity
+    // dbg$4('Fetched worker from %s (%d bytes)', config.scriptSrc, src.length);
+    // const workerBlob = new Blob([src], { type: 'application/javascript' });
+    // const workerUrl = URL.createObjectURL(workerBlob);
     const workerPool = createWorkerPool({
         ...config,
-        scriptSrc: workerUrl,
+        scriptSrc: config.scriptSrc,
     });
     const pool = {
         compileWasm: async (...args) => workerPool.execute('compileWasm', args),
@@ -21314,7 +21321,7 @@ async function createPoolWrapper(config) {
     return {
         ...pool,
         dispose: () => {
-            URL.revokeObjectURL(workerUrl);
+            URL.revokeObjectURL(config.scriptSrc);
             return workerPool.terminate();
         },
     };
